### PR TITLE
Let 0 * real be exactly 0 (fix #1559)

### LINKED ISSUE #1559: Should `0 * 4.0` return `0` or `0.0` ?
Since the operand on the left is exact, the result should be exact, IMO.

cc @mydatamodels

### PR DESCRIPTION
**Pull request by rikardn**
Fix the issue of `integer(0)` times a real number being real 0.0 instead of `integer(0)`. Both `real_double` and `real_mpfr` are fixed.

### CODE REVIEW

File: symengine/real_double.h [hunk 1]
```diff
@@ -228,6 +228,9 @@ class RealDouble : public Number
      * */
     RCP<const Number> mulreal(const Integer &other) const
     {
+        if (other.is_zero()) {
```
Comment (isuruf) [inline symengine/real_double.h:231]:
```suggestion
        if (other.is_exact_zero()) {
```

Comment (rikardn) [reply symengine/real_double.h:231]: We already know it is exact since it is `Integer` so `is_zero` should be enough.

Comment (isuruf) [reply symengine/real_double.h:231]: Yes, you are right.

**Review by isuruf — APPROVED** [review]

**Comment by rikardn** [comment]
Thanks @isuruf for reviewing

**Pull request merged**